backend/cfg/llvmize.ml

```diff
@@ -1217,57 +1252,113 @@ let cfg (cl : CL.t) =
   let fun_ret_type = make_ret_type (Array.to_list fun_ret_type) in
   F.define t ~fun_name ~fun_args ~fun_ret_type ~fun_dbg ~fun_attrs pp_body
 
-(* CR yusumez: Implement this *)
-let data ds =
+(* CR yusumez: We make some assumptions about the structure of the data items we
+   receive and decode it manually here. Ideally, [data_item]s would be
+   represented in a more structured manner, but this should do for now. *)
+let make_temp_data_symbol =
+  let idx = ref 0 in
+  fun () ->
+    let module_name =
+      Compilation_unit.(get_current_or_dummy () |> name |> Name.to_string)
+    in
+    let res = Format.asprintf ".temp.%s.%d" module_name !idx in
+    idx := !idx + 1;
+    res
+
+let data (ds : Cmm.data_item list) =
   let t = get_current_compilation_unit "data" in
-  t.data <- List.append t.data ds
+  let define_symbol' ~private_ ~header ~symbol contents =
+    let symbol =
+      match symbol with
+      | None -> make_temp_data_symbol ()
+      | Some symbol -> symbol
+    in
+    (match header with
+    | None -> ()
+    | Some header ->
+      let header_sym = ".header." ^ symbol in
+      F.data_decl ~private_:true t header_sym [Cint header]);
+    F.data_decl ~private_ t symbol contents;
+    t.defined_symbols <- String.Set.add symbol t.defined_symbols;
+    List.iter
+      (fun (d : Cmm.data_item) ->
+        match[@warning "-4"] d with
+        | Csymbol_address { sym_name; sym_global = _ } ->
+          t.referenced_symbols <- String.Set.add sym_name t.referenced_symbols
+        | _ -> ())
+      contents
+  in
+  let define_symbol ~private_ ~header ~symbol contents =
+    if private_ && List.length contents = 0
+    then () (* No need to declare a private symbol with no contents *)
+    else define_symbol' ~private_ ~header ~symbol contents
+  in
+  let peek_int = function[@warning "-4"] Cmm.Cint n -> Some n | _ -> None in
```

Comment:
(Made me remember that this file
should be added to [1] to be sure
we always format the file. Should
probably be in a dedicated pull
request.)

[1] https://github.com/oxcaml/oxcaml/blob/main/backend/.ocamlformat-enable